packages/app-utils/src/components/ConnectionBootstrap.tsx

```diff
@@ -205,7 +174,50 @@ export function ConnectionBootstrap({
   return (
     <ConnectionContext.Provider value={connection ?? null}>
       <ObjectFetcherContext.Provider value={objectFetcher}>
-        {children}
+        {(() => {
+          if (isShutdown) {
+            return (
```

Comment:
You don't need an IIFE (immediately invoked function expresssion) here. Instead you can just do this. The modals don't show unless `isOpen` is true, so it's fine to mount them here.

```ts
if (isShutdown) {
  return <LoadingOverlay />
}

return (
  <>
    {children}
    <DebouncedModal isOpen={isReconnecting} />
    <BasicModal isOpen={isAuthFailed />
  </>
)
```

If you want conditional rendering inside of JSX, you can do this (just for future reference)

```ts
return (
  {children}
  {isReconnecting && <DebouncedModal isOpen={isReconnecting} />
)
```

But that's not needed here since the modal won't show if `isOpen` is false.